modules/library/jdbc/src/main/java/org/geotools/data/jdbc/FilterToSQL.java

```diff
@@ -746,6 +770,21 @@ protected void visitBinaryComparisonOperator(BinaryComparisonOperator filter, Ob
 
         Expression left = filter.getExpression1();
         Expression right = filter.getExpression2();
+
+        // shortcut for IN case, to have a more natural encoding for the common/sane case of 
+        // "in(property, v1, v2, ... vn) = true" into "property in (v1, v2, ..., vn) 
+        if ("=".equals(extraData) || "!=".equals(extraData)) {
```

Comment:
This should not also check if ``inEncodeEnable`` is set to ``TRUE`` ? As far I can see ``encodeInComparison`` will directly encode the IN clause. I'm sure I miss something but is not clear how we differentiate for the use case were we need to encode or not the IN SQL caluse.

Reply:
Good catch